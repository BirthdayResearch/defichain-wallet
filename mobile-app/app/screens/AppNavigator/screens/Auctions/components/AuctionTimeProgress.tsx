import { View } from '@components'
import { ThemedText } from '@components/themed'
import { useThemeContext } from '@shared-contexts/ThemeProvider'
import { getColor, tailwind } from '@tailwind'
import { translate } from '@translations'

import { useAuctionTime } from '../hooks/AuctionTimeLeft'
import * as Progress from 'react-native-progress'
import BigNumber from 'bignumber.js'
import { StyleProp, TextStyle } from 'react-native'

interface AuctionTimeProgressProps {
  liquidationHeight: number
  blockCount: number
  label: string
  auctionTextStyle?: StyleProp<TextStyle>
}

export function AuctionTimeProgress (props: AuctionTimeProgressProps): JSX.Element {
  const { isLight } = useThemeContext()
  const { timeRemaining, blocksRemaining, blocksPerAuction } = useAuctionTime(props.liquidationHeight, props.blockCount)
  const normalizedBlocks = new BigNumber(blocksRemaining).dividedBy(blocksPerAuction).toNumber()

  return (
    <>
      <View style={tailwind('flex-row w-full justify-between mb-2')}>
        <View style={tailwind('w-6/12')}>
          <ThemedText
            light={tailwind('text-gray-500')}
            dark={tailwind('text-gray-400')}
            style={tailwind('text-xs')}
          >
            {translate('components/AuctionTimeProgress', props.label)}
          </ThemedText>
        </View>
        <View style={tailwind('flex flex-row flex-1 justify-end flex-wrap')}>
          <ThemedText
            light={tailwind('text-gray-900')}
            dark={tailwind('text-gray-50')}
            style={[tailwind('text-sm text-right'), props.auctionTextStyle]}
          >
            {timeRemaining !== '' && translate('components/AuctionTimeProgress', '~{{time}} left', { time: timeRemaining })} ({translate('components/AuctionTimeProgress', '{{block}} blks', { block: blocksRemaining })})
          </ThemedText>
        </View>
      </View>
      <Progress.Bar
        progress={normalizedBlocks}
        width={null}
        borderColor={getColor(isLight ? 'gray-300' : 'gray-600')}
        color={getColor(isLight ? 'blue-500' : 'blue-500')}
        unfilledColor='gray-100'
        borderRadius={8}
        height={8}
      />
    </>
  )
}
