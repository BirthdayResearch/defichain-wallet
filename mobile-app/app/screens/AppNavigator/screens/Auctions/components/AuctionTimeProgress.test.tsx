import { render } from '@testing-library/react-native'
import { AuctionTimeProgress } from './AuctionTimeProgress'

jest.mock('@shared-contexts/ThemeProvider')
jest.mock('@shared-contexts/NetworkContext')

describe('Auction time progress', () => {
  it('should match snapshot', async () => {
    const liquidationHeight = 1100
    const blockCount = 1000
    const label = 'Auction time remaining'
    const rendered = render(
      <AuctionTimeProgress
        liquidationHeight={liquidationHeight}
        blockCount={blockCount}
        label={label}
      />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
