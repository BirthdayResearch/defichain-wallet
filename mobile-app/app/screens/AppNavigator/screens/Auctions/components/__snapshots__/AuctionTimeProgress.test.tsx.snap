// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auction time progress should match snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 12,
                "lineHeight": 16,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
      >
        Auction time remaining
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexBasis": "0%",
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Array [
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                  "textAlign": "right",
                },
                undefined,
              ],
              Object {
                "color": "rgba(23, 23, 23, 1)",
              },
            ],
          ]
        }
      >
        ~50m left
         (
        100 blks
        )
      </Text>
    </View>
  </View>,
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    onLayout={[Function]}
    progress={2.7777777777777777}
    style={
      Array [
        Object {
          "backgroundColor": "gray-100",
          "borderColor": "rgba(212, 212, 212, 1)",
          "borderRadius": 8,
          "borderWidth": 1,
          "overflow": "hidden",
          "width": null,
        },
        undefined,
      ]
    }
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(90, 169, 230, 1)",
          "height": 8,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "scaleX": 1,
            },
          ],
        }
      }
    />
  </View>,
]
`;
